bulk-install-app/install.py

```diff
@@ -76,42 +73,34 @@ def process_install(session, authenticity_token, target_id, target_type, version
     }
 
     installed_response = session.post(url, installation_request_data, headers=headers)
-    installed_page_soup = BeautifulSoup(installed_response.content, 'html.parser')
     save_debug_info(folder='install-data', name='installed', response=installed_response)
 
-    return installed_page_soup
+    return installed_response
 
 
-def redirect_install(session, installed_page_soup) -> bool:
+def check_install_result(installed_response) -> bool:
     """
     Handle the redirect to the app's callback URL.
     Returns a boolean for whether the installation succeeded.
     """
-    redirect_element = installed_page_soup.find('a', {'id': 'redirect'})
-    if not redirect_element:
-        return False
-
-    redirect_url = redirect_element.get('href')
-    if not redirect_url:
-        return False
-
-    redirect_response = session.get(redirect_url)
-    save_debug_info(folder='install-data', name='redirect', response=redirect_response)
-    return 'success' in redirect_response.text.lower()
+    save_debug_info(folder='install-data', name='check', response=installed_response)
+    return 'was installed on' in installed_response.text.lower()
 
 
-def install(session, target_name, target_id) -> Optional[bool]:
+def install(session, target_name, target_url) -> Optional[bool]:
     """
     Manages full installation workflow - start, process, redirect.
 
     Returns boolean for whether installation has occurred without any errors.
     Returns None if user did not have permissions to install, or if errors occurred in pre-installation parsing.
     """
+    target_id = parse_qs(urlparse(target_url).query)['target_id'][0]
```

Comment:
nit: Can we move this into the `try` block just to make sure that any exception that is raised will be handled?